chocs_middleware/cache/middleware.py

```diff
@@ -100,22 +112,31 @@ def handle(self, request: HttpRequest, next: MiddlewareHandler) -> HttpResponse:
 
         if "vary" not in response.headers:
             response.headers["vary"] = ",".join(cache_vary)
-        else:
```

Comment:
Could it be the case that "vary" is in response.headers but "etag" is not?
I wonder if we would still need to generate a new cache_id in this case?